gapis/api/gles/state.go

```diff
@@ -20,55 +20,90 @@ import (
 	"github.com/google/gapid/gapis/api"
 )
 
+// fbai is the result getFramebufferAttachmentInfo.
+type fbai struct {
+	width        uint32
+	height       uint32
+	format       GLenum // sized
+	multisampled bool
+}
+
+func attachmentToEnum(a api.FramebufferAttachment) (GLenum, error) {
+	switch a {
+	case api.FramebufferAttachment_Color0:
```

Comment:
So you know what although the enums are finite, the number of attachments is, in theory, unbounded. (implementation must support at least 4)

So ideally this should handle Depth and Stencil and do cast and arithmetic for the rest.

Alternatively what do you think of making FramebufferAttachment an integer instead?  (where >= 0 is the respective colour, and negative values are used for depth/stencil)

Reply:
Understood, but these are declared in the protobuf. [Negative enums in protos are discouraged](https://developers.google.com/protocol-buffers/docs/proto#enum).
I'll have a think about this, but this issue has not been made better or worse by this PR - opting to defer.

Reply:
How about low two bits indicate type?

I am kind ok fine with the current approach, as well.  But let's maybe add a few more?

Reply:
If you want more changes, let's do it in a follow up CL. This PR is urgent.

Reply:
Since extending the Enum would require changes to the UI as well, that is probably best done later?

The other changes look fine to me.